aee.js

```diff
@@ -82,3 +82,19 @@ Array.prototype.count = function (spec) {
   }
   return result
 }
+
+Array.prototype.max = function (comparer) {
+  const len = this.length
+  if (len === 0) return null
+  let result = this[0]
+  let comparerSpec = comparer
+  if (typeof comparerSpec !== 'function') {
+    comparerSpec = (!isNaN(this[0]) || this[0].constructor === Object)
```

Comment:
I decided it should throw an error, as there's no point on applying `max()` on objects without the comparer, but writing the test (and then complying with it) has been troublesome.